### PR TITLE
Adds image push to internal registry in the context of OpenShift

### LINKED ISSUE #2250: Push Images directly into Harbor in the context of Openshift
Currently Lagoon uses the internal registry of OS for images.

However, without the images being pushed to Harbor, OS contexts don't benefit from the Lagoon Insights image scanning.

Having the resulting images also pushed into Harbor would allow us to run scans (even though the images are pulled from the internal OS registry).

### PR DESCRIPTION
**Pull request by bomoko**
- [*] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [*] PR title is ready for changelog and subsystem label(s) applied

As described in #2250 - having the images built within the context of Openshift _also_ inside of Harbor will allow us to run vulnerability scans on those images automatically.

This PR logs into the internal registry (i.e. Harbor), then pushes the resulting images into the internal registry using similar logic to the standard image push.

# Closing issues

closes #2250

### CODE REVIEW

File: images/oc-build-deploy-dind/build-deploy-docker-compose.sh [hunk 1]
```diff
@@ -961,6 +961,11 @@ elif [ "$TYPE" == "pullrequest" ] || [ "$TYPE" == "branch" ]; then
     parallel --retries 4 < /oc-build-deploy/lagoon/push
   fi
 
+  # If we have Images to Push to the the Internal Harbor, let's do so
+  if [ -f /oc-build-deploy/lagoon/push_harbor ]; then
```
Comment (Schnitzel) [inline images/oc-build-deploy-dind/build-deploy-docker-compose.sh]: can you run them also with `--retries 4`? just to be sure they actually end up in harbor :)

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: images/oc-build-deploy-dind/build-deploy-docker-compose.sh (hunk 1 above)
Comment (bomoko) [reply images/oc-build-deploy-dind/build-deploy-docker-compose.sh]: Done! :)

File: images/oc-build-deploy-dind/build-deploy.sh [hunk 2]
```diff
@@ -40,6 +40,10 @@ DOCKER_REGISTRY_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
 
 docker login -u=jenkins -p="${DOCKER_REGISTRY_TOKEN}" ${OPENSHIFT_REGISTRY}
 
+if [ ! -z ${INTERNAL_REGISTRY_URL} ] && [ ! -z ${INTERNAL_REGISTRY_USERNAME} ] && [ ! -z ${INTERNAL_REGISTRY_PASSWORD} ] ; then
```
Comment (Schnitzel) [inline images/oc-build-deploy-dind/build-deploy.sh:50]: unfortunately the variables you check here are never created, we need to parse them out of the project variables out first, see the code in kubectl-build-deploy-dind

https://github.com/amazeeio/lagoon/blob/a9807db86b9fbe0826a7211f2322303ad67bf362/images/kubectl-build-deploy-dind/build-deploy.sh#L11-L15

Comment (bomoko) [reply images/oc-build-deploy-dind/build-deploy.sh:50]: Cool - changed!

File: images/oc-build-deploy-dind/build-deploy.sh [hunk 3]
```diff
@@ -40,6 +46,14 @@ DOCKER_REGISTRY_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
 
 docker login -u=jenkins -p="${DOCKER_REGISTRY_TOKEN}" ${OPENSHIFT_REGISTRY}
 
+INTERNAL_REGISTRY_LOGGED_IN="false"
+if [ ! -z ${INTERNAL_REGISTRY_URL} ] && [ ! -z ${INTERNAL_REGISTRY_USERNAME} ] && [ ! -z ${INTERNAL_REGISTRY_PASSWORD} ] ; then
+  echo "docker login -u '${INTERNAL_REGISTRY_USERNAME}' -p '${INTERNAL_REGISTRY_PASSWORD}' ${INTERNAL_REGISTRY_URL}" | /bin/bash
+  if ["$?" -eq 0 ] ; then
```
Comment (Schnitzel) [inline images/oc-build-deploy-dind/build-deploy.sh]: missing space :)

**Review by Schnitzel — APPROVED** [review]
tested via deploying a drupal-example and all worked as it should! well done!!

**Pull request merged**